q2_types/feature_data/_format.py

```diff
@@ -161,6 +162,11 @@ def _validate_lines(self, max_lines):
                             raise ValidationError('Multiple consecutive IDs '
                                                   'starting on line '
                                                   f'{line_number-1!r}')
+                        if line in ids:
```

Comment:
One problem with this is that it treats the entire header line as the ID. I believe that we need to treat everything between the `>` and the first instance of whitespace (` `, if any) as the ID. Characters after the space are part of the record's description field, and are not subject to the ID uniqueness rules.

@ebolyen, can you please confirm this for me?

Reply:
Yep, that is correct.

Reply:
Thanks @ebolyen! @Oddant1, now would be a good time to fix the error message on https://github.com/qiime2/q2-types/pull/220/files#diff-c7c5016b403f1d6d2fb45764b58ae9d6R153 to indicate this situation, too (technically the ID doesn't start with `>`, the _refline_ for the record starts with `>`.

Reply:
Are they allowed to have a space immediately after the `>`? We don't currently check for that at all.

Reply:
I _think_ the most comprehensive resource is here:

http://scikit-bio.org/docs/0.5.5/generated/skbio.io.format.fasta.html#module-skbio.io.format.fasta

Have fun... :tada:

Reply:
I guess if they had a space immediately after the `>` their id would start with a space and that would be silly, so I suppose they can't do that

Reply:
I would probably interpret it as a "null" ID, but in any case, that is almost certainly a formatting error and we should probably avoid accepting it.